### PR TITLE
Feature/TCSOB-554-button-component

### PR DESCRIPTION
**Pull request by lyoon-tcs**
<h1>Background</h1>
Create a simple Button Component, with proper tests using React Testing Library. Focus is keeping component small, practice tests for individual component. Focus is more on user (they should be able to see the button), rather than things they won't care about (a button should be a color based on the className added through props).
<h1>Breaking Changes</h1>
N/A
<h1>Updates</h1>
A button component, with passing tests.
<h1>Example Links</h1>
N/A
<h1>Key Files</h1>
Button.js, Button.test.js

### CODE REVIEW

**Comment by lyoon-tcs** [comment]
First comment - hello Ariana! Not sure what conflicts are/how to resolve them. Read about them, got a brief understanding - in our 1-on-1 can we go through this together?

**Comment by lyoon-tcs** [comment]
Meanwhile I'll work on getting more practice, creating and testing (individually) small components.

**Review by achang-tcs — CHANGES_REQUESTED** [review]
Nice job with your first PR @lyoon-tcs ! Please see my feedback below:

- Add source code copyright at the top of file for all JS files (`App.js`, `Button.js`, `Button.test.js`)
- Add a comment with a function description (`@description, @param, @returns`) for the button element.
- See additional feedback left as in-line comments in the `Button.test.js` file.

File: src/components/tests/Button.test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import Button from '../Button';
+
+afterEach(() => {
+  cleanup();
+});
+
+describe('Button Component renders correctly', () => {
+  const buttonText = 'hello world';
+  render(<Button buttonText={buttonText} />);
+  screen.debug();
```
Comment (achang-tcs) [inline src/components/tests/Button.test.js]: remove this line if it is only used for local debugging

File: src/components/tests/Button.test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { render, screen, cleanup } from '@testing-library/react';
+// local files
+import Button from '../Button';
+
+afterEach(() => {
+  cleanup();
+});
+
+describe('Button Component renders correctly', () => {
+  const buttonText = 'hello world';
```
Comment (achang-tcs) [inline src/components/tests/Button.test.js:22]: add lines 13-16 directly in the `test('should render button with proper text')` block, as they are not used for the snapshot. variables/implementation should be relative to the function block where they are used.

File: src/components/tests/Button.test.js (hunk 1 above)
Comment (lyoon-tcs) [reply src/components/tests/Button.test.js]: Done

File: src/components/tests/Button.test.js (hunk 2 above)
Comment (lyoon-tcs) [reply src/components/tests/Button.test.js:22]: For this one, moved everything except buttonText, as the snapshot uses this variable. Let me know if this is okay - thanks

**Review by achang-tcs — APPROVED** [review]
nice job applying code review feedback! OK to merge 👍

**Pull request merged**